app/routes/api/users.js

```diff
@@ -32,7 +32,12 @@ router.get('/:username', function(req, res, next){
 });
 
 // create new user and authenticate with passport
+
 router.post('/', function(req, res, next){
+  if(req.body.password != req.body.password2){
+    res.json(err);
+  }
```

Comment:
You need to enclose the reset of the function in an else block, because even if res.json is called, the rest of the function will still run (Really stupid, I know :-1: ).